CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.12)
```

Comment:
Minimum version for the new [FindPython](https://cmake.org/cmake/help/v3.12/module/FindPython.html) module in CMake

Reply:
Is there a possibility to check if the current cmake version is >= 3.12 then use the new FindPython, otherwise default to the old way (with maybe a warning)?

I don't want to break anyone's build

Reply:
fixed